### ISSUE TITLE
Using memory mode map causes a crash with streaming

### ISSUE THREAD

**Issue by amoss** [netdata/netdata#9376]
##### Bug report summary

* Setup 45 child nodes in docker containers (the config is in `build_external/scenarios/gaps_hi`), can be auto-scaled by docker.
* Connect to a parent node running `memory mode = save`
* Shutdown / restart the child nodes.
* Parent node crashes because `callocz()` fails to allocate memory.

##### OS / Environment
Debian 10

##### Netdata version
netdata v1.22.1-306-g8186870c

##### Component Name
streaming

##### Steps To Reproduce
In summary

##### Expected behavior
Should allocate more memory than that (crashes at about 300MB usage). Should not crash.

**Comment by amoss** [comment]
I had forgotten about this. I'll setup a test and see what happens.

**Comment by cakrit** [comment]
@mfundul plz check you have some time.
